catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.java

```diff
@@ -113,6 +119,14 @@ protected void onCreate(Bundle savedInstanceState) {
 			return;
 		}
 
+		if (getIntent().getBooleanExtra(RESTART_INTENT, false)) {
+			Intent restartIntent = getApplicationContext().getPackageManager().getLaunchIntentForPackage(getApplicationContext().getPackageName());
+			PendingIntent intent = PendingIntent.getActivity(getApplicationContext(), 0, restartIntent, PendingIntent.FLAG_CANCEL_CURRENT);
+			AlarmManager manager = (AlarmManager) getApplicationContext().getSystemService(Context.ALARM_SERVICE);
+			manager.set(AlarmManager.RTC, System.currentTimeMillis() + restartDelayAfterAppClose, intent);
```

Comment:
i cannot. The 500ms seemed to be plenty of time on the tested devices/emulators, but i can't guarantee that it will work on any device out there.